### DISCUSSION TITLE
RocketMQ 多集群同步问题

### DISCUSSION THREAD

**Discussion by pxd98** [apache/rocketmq#9053] [Q&A]
我想问一下RocketMQ有多集群同步工具吗？就是搭建了两个RocketMQ集群，然后生产者和消费者只往一个集群上发送，但是希望这个集群中的消息可以同步到另一个集群上。
我想到的一个变通方法是这样子的第一个集群中的节点注册到这个集群的namespace上，另一个集群中的节点除了注册到自己的namespace上，还会以从节点的身份注册到第一个集群的namespace上。

**Comment by humkum** [comment]
同一个 namesrv 下的不同 broker 不会同步数据的，你这样注册到第一个集群的 namesrv 上，只会将不同的消息发送到两个集群的 broker，两个集群的消息不是相同的，而是两个集群的消息加在一起是全量消息。

对于这种情况，建议自己写一个工具，内部实现一个第一个集群的 consumer，消费对应的 topic 消息，然后用第二个集群的 producer 发送到第二个集群。

**Reply by onejimmyboy** [reply]
没有开源的同步工具吗？
